### PR TITLE
Add lint and CLA workflows

### PR DESCRIPTION
**Pull request by tgolen**
This adds basic GH workflows to ensure that code is formatted properly before it is merged into the repo.

1. Linting - this applies all of the same linting rules as Expensify/App
2. Prettier - this applies all of the same prettier rules as Expensify/App

### CODE REVIEW

**Comment by tgolen** [comment]
@neil-marcellini Can you please review and merge this?

**Comment by neil-marcellini** [comment]
:eyes taking a look!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<!-- Explanation of the change or anything fishy that is going on -->
+
+### Related Issues
+<!-- Please replace GH_LINK with the link to the GitHub issue this Pull Request is related to -->
+GH_LINK
+
+### Automated Tests
+<!---
+Most changes to Onyx should have accompanying tests. Describe the tests you added or if no tests were added an explanation about why one was not needed.
+--->
```
Comment (neil-marcellini) [inline .github/PULL_REQUEST_TEMPLATE.md]: Let's change this to `### Manual Tests` please and remove the mention of Onyx

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                  key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+                  restore-keys: |
+                      ${{ runner.os }}-node-
+
+            - name: Setup Node
+              uses: actions/setup-node@v1
+              with:
+                  node-version: "16.x"
+
+            - run: npm ci
```
Comment (neil-marcellini) [inline .github/workflows/lint.yml:25]: Does this work without a package.json in the repo?

File: .github/workflows/lint.yml
Comment (neil-marcellini) [inline .github/workflows/lint.yml:1]: This workflow is a bit different than the one in the App. Why?

**Review by neil-marcellini — CHANGES_REQUESTED** [review]
One suggestion and a few questions. How can I test this?

File: .github/workflows/lint.yml
Comment (tgolen) [reply .github/workflows/lint.yml:1]: I based it off of the workflow from React-Native-Onyx actually, and just copied the prettier command from App. 

The workflow in App looks like it uses an additional workflow `Expensify/App/.github/actions/composite/setupNode@main` which probably helps to make it DRY and isn't something we need here.

The workflow in App also looks at shell scripts and lints them which isn't something we need in this repo.

I think it's fine to have it match what we have in React-Native-Onyx for now.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (tgolen) [reply .github/workflows/lint.yml:25]: Probably not, but I committed the `package.json` directly to `main` (oops) so it's already there.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (tgolen) [reply .github/PULL_REQUEST_TEMPLATE.md]: Ah, good call. I didn't audit this at all.

**Comment by tgolen** [comment]
> How can I test this?

You can test it by pushing changes to this branch to `index.js` that has ESLint errors. However, I already did that test as you can see in the commit log, so it's not really necessary to do again.

**Comment by tgolen** [comment]
Updated the verbiage in the PR template

**Review by neil-marcellini — APPROVED** [review]
Ok I think this is good to go.

**Pull request merged**